client/rest-high-level/src/test/java/org/elasticsearch/client/ESRestHighLevelClientTestCase.java

```diff
@@ -153,4 +153,9 @@ protected Settings restClientSettings() {
             .put(ThreadContext.PREFIX + ".Authorization", token)
             .build();
     }
+
+    @Override
+    protected boolean getStrictDeprecationMode() {
```

Comment:
I would love to! But in the mean time #34135 got merged and it added a deprecated setting:

https://github.com/elastic/elasticsearch/blob/55eaf7a3ff83d032fd4f2263e144113909d1b2bd/client/rest-high-level/build.gradle#L93

I think this is the reason for the two tests below to fail once the strict deprecation mode is on: 
```
./gradlew :client:rest-high-level:integTestRunner -Dtests.seed=B82DCBDDD63664E9 -Dtests.class=org.elasticsearch.client.documentation.ClusterClientDocumentationIT -Dtests.method="testClusterGetSettings" 

./gradlew :client:rest-high-level:integTestRunner -Dtests.seed=B82DCBDDD63664E9 -Dtests.class=org.elasticsearch.client.ClusterClientIT -Dtests.method="testClusterGetSettingsWithDefault"
```
```
[xpack.security.transport.ssl.truststore.password] setting was deprecated in Elasticsearch and will be removed in a future release! See the breaking changes documentation for the next major version.
```
I did not see what setting to use instead of `xpack.security.transport.ssl.truststore.password`. TBH I am not really sure if the setting should be deprecated in the first place as I did not find it mentioned in any docs...

@nik9000 do you have any ideas / pointers?

Reply:
> I would love to! But in the mean time #34135 got merged and it added a deprecated setting

@jkakavas, can you comment on this setting being deprecated?

Reply:
Yes, it is deprecated
https://github.com/elastic/elasticsearch/blob/46c7b5ee6e53e33aa6f91c1cbd873810d319a400/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SSLConfigurationSettings.java#L92-L95

and I shouldn't have used it. I'll raise a follow-up PR tomorrow my morning to address this.

Reply:
:heart: